CoffeeShop/src/main/java/order/OrderList.java

```
@@ -79,7 +88,7 @@ public Boolean add(Order order) throws InvalidOrderException {
      * @param ID The ID used to find the order to be removed
      */
     @Override
-    public Boolean remove(UUID ID) {
+    public synchronized boolean remove(UUID ID) {
```

I know I should have suggested sooner (not on this PR, sorry) but I think we can refactor this remove method to go like this to let the exception bubble up just because we can't really do much if it errors here

```
public synchronized boolean remove(UUID ID) throws InvalidOrderException {
    completeOrders.add(this.getOrder(ID));

    return inCompleteOrders.removeIf(order -> order.getOrderID().equals(ID));
}
```